### ISSUE TITLE
COMBAT_LOG_EVENT_UNFILTERED bugs while "Advanced Combat Logging" is enabled in the options panel

### ISSUE THREAD

**Issue by Tercioo** [Stanzilla/WoWUIBugs#326]
I have users reported and tested myself a couple of times:

While "Advanced Combat Logging" is turned OFF, the registered combatlog.txt file (with the command /combatlog) and the ingame combat log provided by "COMBAT_LOG_EVENT_UNFILTERED" matches.

While "Advanced Combat Logging" is turned ON, the ingame "COMBAT_LOG_EVENT_UNFILTERED" has around 10% less events, tested the damage part of the logs, all spells, melee, ranged, pets seems to be affected.

**Comment by Tercioo** [comment]
I don't have a "how to reproduce" this issue other them "play it with then without advanced combat logging", so this two images comparing combatlog.txt and ingame cleu.
The first image (left) show the combatlog.txt with 17 casts and 17 spell damage.
The second image show ingame CLEU with 17 casts of a spell but only 15 damage events.
Spell is Cobra Shot on boss Lord Chamberlain, last boss of Halls of Attonement, the boss has no absorb shields, timestamp on ingame cleu converted.

![cleu2](https://user-images.githubusercontent.com/9000280/199141375-d54aa8dc-fa2b-489f-b996-e1cba01c126f.png)![cleu](https://user-images.githubusercontent.com/9000280/199141392-35735db2-a87b-49dd-bd8a-52ed314f6c58.png)

**Comment by MysticalOS** [comment]
Just adding but saw this bug MANY times during raid testing. SPELL_CAST_START events for boss abilities would randomly be missing for the person that was raid logging with advanced combat logging. At the time we thought the distance filter for raids got incorrectly re-enabled (raids are supposed to have unlimited CLEU distance to avoid range issues).

Turns out, that likely wasn't cause at all, this was.

**Comment by Flamanis** [comment]
Can confirm that somehow events are getting eaten while Advanced Combat Logging is enabled. Ran two m+ +2, me and one other saved all the game CLEU to a saved variable to export it, compared to the log generated by the game from that players machine.

First run with neither player having it enabled, had about 40ish events in the combatlog that weren't from CLEU. Pretty much just the regular stuff like encounter/emote/challengemode that aren't in CLEU anyways. 

Second run, had one of us running with it enabled and the other running disabled.

![cmd_2022-11-03_19-11-46](https://user-images.githubusercontent.com/5752056/199859681-0d103a1c-5bd2-4cd5-9f6d-bb97f8a19ff6.png)
![cmd_2022-11-03_19-11-59](https://user-images.githubusercontent.com/5752056/199859690-c63022cc-a0dc-4adf-9cf3-cec1ce833fd4.png)


Uploaded both logs through WCL to parse them, same damage totals as the one with it disabled. 

Even not counting Swing_Damage_Landed which iirc doesn't hit CLEU normally anyways, that's almost 1800 damage/heal events just disappeared.

**Comment by funkydude** [comment]
> that's almost 1800 damage/heal events just disappeared

Did you notice any SPELL_CAST_START events missing from NPCs?

**Comment by Flamanis** [comment]
As the screenshot shows. 16 were missing from that run. Which ones specifically, I did not check. I only checked events by token.

**Comment by kvigue** [comment]
The WoW Team has identified an issue here and plans to fix it in patch 10.0.2. Thank you for the report and extra testing information; it was helpful! Please retest once 10.0.2 is out and let us know if you still see issues.

**Comment by Flamanis** [comment]
Still seeing this behavior on live 10.0.2. 

First two images are the Non Advanced IN GAME, compared to both versions of the combatlog.txt file. As you can see, when the Advanced .txt is enabled, the majority of different events are SWING_DAMAGE_LANDED, so it is accurate.
![cmd_2022-11-15_17-55-16](https://user-images.githubusercontent.com/5752056/202051782-004672c6-4765-40ea-baa9-8256b2ea2dff.png)
![cmd_2022-11-15_17-55-26](https://user-images.githubusercontent.com/5752056/202051788-e1c82652-16f5-4c99-b2a4-e9c944a5906d.png)


Second set of images is with Advanced Enabled IN GAME, compared to both versions of the combatlog.txt file. The only difference between the two remains to be SWING_DAMAGE_LANDED (And the SPELL_CAST_FAILED cause I spam buttons). 


![cmd_2022-11-15_17-54-54](https://user-images.githubusercontent.com/5752056/202052013-1a9588b9-0a1a-4cac-a5a6-18f650aea79d.png)
![cmd_2022-11-15_17-55-07](https://user-images.githubusercontent.com/5752056/202052017-9e381a29-50cf-4b56-9d93-933c793709e2.png)

Tl;Dr, Advanced Combat Logging is still causing events to be dropped.

**Comment by kvigue** [comment]
Thanks for the heads up! Good news is that I just made a human error of overpromising on the timeline... The change was too late for the 10.0.2 release candidate, and the hotfix for it hasn't been fully deployed yet. (I had been hoping the hotfix would get deployed in time for launch, but it didn't make it through the queue.)

I'll try to ensure this gets a public hotfix note when it goes out, which will hopefully be sometime in the next few weeks. Sorry for the bad information, and thanks for checking and keeping me honest!

**Comment by Tercioo** [comment]
@kvigue a blue post confirming combat log is bugged would be a nice thing to do as now players are confused about their damage done.

**Comment by Flamanis** [comment]
Ran a couple m0s and checked using the same process as above. Looks to be fixed. Event token counts were lockstep.


I'll do intermitent testing during next week and when m+ releases. But so far, I think it's good.

**Comment by MysticalOS** [comment]
Yeah I think this was silently fixed with no update, I had someone else do a test earlier today and could no longer repo issue

**Comment by kvigue** [comment]
Yep, apologies for not following up. The fix should be deployed in any builds >=46781.

Thanks for retesting; the confirmation is appreciated!

**Comment by funkydude** [comment]
> Yep, apologies for not following up. The fix should be deployed in any builds >=46781.
> 
> Thanks for retesting; the confirmation is appreciated!

Thanks for the details, we appreciate it!
